README.md

```diff
@@ -2,115 +2,119 @@
-
-The library has been extensively tested with Hypothesis. For more details on running the tests, refer to this [section](#run-tests).
-
-The design of this library was originally inspired by Influenceth's [64x61 fixed-point math library](https://github.com/influenceth/cairo-math-64x61).
+This library provides a felt-based implementation of arithmetic functions for two types of fixed-point decimal numbers, `Wad` (18 decimals of precision) and `Ray` (27 decimals of decimal numbers), written in Cairo for [StarkNet](https://docs.cairo-lang.org/index.html).
 
 ## Overview
 
-This library includes arithmetic, aggregation, conversion and bounds check functions.
+This library includes arithmetic, comparison and conversion functions.
 
 ### Arithmetic
 
 #### Addition and Subtraction
 
-These functions operate on both `wad` and `ray`.
+Both `Wad` and `Ray` implement the `Add`, `AddEq`, `Sub` and `SubEq` traits.
 
-- `add(a, b)`: Addition `a` and `b` with a bounds check that the result is within the range [-2<sup>125</sup>, 2<sup>125</sup>]
-- `unsigned_add(a, b)`: Unsigned addition of `a` and `b` with a bounds check that the result is within the range [0, 2<sup>125</sup>]
-- `sub(a, b)`: Subtraction of `b` from `a` with a bounds check that the result is within the range [-2<sup>125</sup>, 2<sup>125</sup>]
-- `unsigned_sub(a, b)`: Unsigned subtraction of `a` and `b` with a bounds check that the result is within the range [0, 2<sup>125</sup>]
+Addition and subtraction can be performed as follows:
+- `a + b`
+- `a += b`
+- `a - b`
+- `a -= b`
+where both `a` and `b` are of the same type.
 
 #### Multiplication and Division
 
-The prefixes `w` and `r` denotes whether the function operates on `wad` or `ray` respectively.
+Both `Wad` and `Ray` implement the `Mul`, `MulEq`, `Div` and `DivEq` traits.
+
+Multiplication and division can be performed as follows:
+- `a * b`
+- `a *= b`
+- `a / b`
+- `a /= b`
+where both `a` and `b` are of the same type.
+
+There is also a set of functions for operations involving a `Wad` and a `Ray`:
+- `wmul_wr`/`wmul_rw`: Multiply a `Wad` value with a `Ray` value, and divide the product by one `Wad` scale to return a `Ray`
+- `wdiv_rw`: Scale up a `Ray` value by one `Wad` scale and divide the scaled value by a `Wad` value to return a `Ray`
+- `rmul_wr`/`rmul_rw`: Multiply a `Wad` value with a `Ray` value, and divide the product by one `Ray` scale to return a `Wad`
+- `rdiv_wr`: Scale up a `Wad` value by one `Ray` scale and divide the scaled value by a `Ray` to return a `Wad`
+- `rdiv_ww`: Scale up a `Wad` value by one `Ray` scale and divide the scaled value by a `Wad` to return a `Ray`
+
+For multiplication, the prefixes `w` and `r` denote whether the product is divided (i.e. scaled down) by a `Wad` or `Ray` respectively. For division, the prefixes `w` and `r` denote whether the first operand is multiplied (i.e. scaled up) by a `Wad` or `Ray` before the division respectively.
```

Comment:
Can you add a note somewhere around here about the loss of precision with some of these operations? Just to be really obvious about it.

![](https://media.giphy.com/media/28n0C19zo9OOvHnYww/giphy.gif)